src/02-room-details.js

```diff
@@ -49,7 +73,35 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  //created an empty arr will collect the rooms that include room id
+  let connectedArr = [];
+  // will retrun the room with the names
+  let newArr = [];
+  // looping throught
+  for (let i = 0; i < rooms.length; i++) {
+    if (rooms[i].roomId === (id)) {
+      // created to house the c
+      connectedArr = (rooms[i].connectsTo);
+    }
+
+  }
+  for (let j = 0; j < rooms.length; j++) {
+    if (connectedArr.includes(rooms[j].roomId)) {
+      newArr.push(rooms[j].name)
+    }
+  }
+  // the array can be empty and there will be no match. connectedArr.length is the saw as id
+    if (!connectedArr.length) {
+    return `Room with ID of '${id}' could not be found.`
+  }
+  // if the room ids are not the same then it must 
+   if (connectedArr.length !== newArr.length) {
+    return `Room with ID of 'incorrect-id' could not be found.`
```

Comment:
this is considered hardcoded. What if the id was `abc1234`, we would still want to get this error message with the incorrect id dynamically inserted in the string.